Use new full text search in search box

yonatann opened the pull request:
Results are either a budget line or an entity.
Currently both look the same. 

@mushon can you create a design of the search box result so that the user can easily see which results are budget lines and which are entities?

akariv commented on open-budget/app/scripts/searchbar.coffee:
```diff
@@ -110,14 +110,14 @@ class BudgetPartitionLayoutView extends Backbone.View
         @cls = (d,suffix="") => window.changeClass( d.value, d.value*(d.o+100)/100.0 ) + suffix
 
         onSuccess = (root) =>
+            if @root
```
why do we need this?

akariv replied:
(or did you mean 'if root' and not 'if @root' ? )

yonatann replied:
Yes it should be 'if root'

It protected from cases that only happen because the local database is still incomplete

yonatann commented:
Fixed the typo

yonatann commented:
Added a fix fo #190 
@mushon please see if the design is what you had in mind.

akariv commented:
This doesn't work... seems like you forgot to push some files, for example:

 - interval_query doesn't have any js in it
 - index.html doesn't include interval_query

yonatann commented:
Yep....
I made the commit not from the root directory and didn't use git submodule for an external repository.

Should work now

akariv commented:
Very nice!
I would put the name and title of the person next to his/her photo (maybe on the left side of the guideline?)

another thing - can you please rebase with upstream master?

yonatann commented:
I've rebased upstream master, its saying that it is already up to date

yonatann commented:
I've also added names and fixed a small bug that caused the tip to flicker

akariv commented:
Looks better :smile: 

Two more little things:

 - Can you add the title somewhere (e.g. 'שר האוצר')? you can use the full_title and put it next to the name.
 - The name tags get cropped when the guideline is moved to the extreme right... weird.

yonatann commented:
I've added the title and fixed the CSS bug.
there is another case where this bug applies and that is when you hover over a participant next to the end of the screen.
I left it for now because it requires some restructuring of the elements

akariv commented:
Where did the minister of culture go? compare:

- http://www.obudget.org/#budget/2027/2014/main
- http://127.0.0.1:9000/#budget/2027/2014/main?tab=changes

yonatann commented:
There is something wrong with the data.
The duplicate detector that I added removed it because culture is attributed to education on the same date range.

![screen shot 2015-06-01 at 1 03 10 pm](https://cloud.githubusercontent.com/assets/1035320/7910633/0f845f46-085f-11e5-901f-4275c8c3322f.png)

akariv commented:
Aha - you should do the dedup according to the title, not the kind.

In short -
 - every budget item is mapped to several kinds (in our case: 2027 ->
['pm', 'education', 'finance' ... ]
 - every kind can have a few titles (in our case 'education' -> ['שר
החינוך', 'שר התרבות']). Each title should its own timeline
 - every item on a timeline can have a slightly different full_title, e.g.
    - 'שרת התרבות והספורט'
    - 'שר החינוך והתרבות'
etc.

On Mon, Jun 1, 2015 at 1:07 PM Yonatan Naor <notifications@github.com>
wrote:

> There is something wrong with the data.
> The duplicate detector that I added removed it because culture is
> attributed to education on the same date range.
>
> [image: screen shot 2015-06-01 at 1 03 10 pm]
> <https://cloud.githubusercontent.com/assets/1035320/7910633/0f845f46-085f-11e5-901f-4275c8c3322f.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenBudget/open-budget-frontend/pull/228#issuecomment-107386763>
> .
>

yonatann commented:
Aha! Makes sense.
Fixed
